### DISCUSSION TITLE
构建代码如何获取umi的history配置

### DISCUSSION THREAD

**Discussion by starsoul666** [umijs/umi#11938] [Q&A]
代码中需要根据history: { type: 'hash' }的type类型生成不同的代码，代码中如何获取history的type配置，类似process.env.node_env变量

**Comment by fz6m** [comment]
你指的是插件来生成不同的代码还是代码中根据 history type 的不同有不同的逻辑。

如果是前者，读 `api.userConfig` 即可，如果是后者，通过 [`define`](https://umijs.org/docs/api/config#define) 自己注入一个环境变量来判断就行。
